In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
DATA = r"C:\Users\kaush\Desktop\fuck_you_chatPAL\pal\datasets\reduced_gsm_with_sub_qs.jsonl"

In [5]:
df = pd.read_json(DATA, lines=True)

In [7]:
df.columns

Index(['input', 'target', 'prev_answer', 'prev_score', 'prev_generation',
       'prev_code_error', 'subquestions'],
      dtype='object')

In [11]:
for col in df.columns:
    print(df[col][0])

Gloria is shoe shopping when she comes across a pair of boots that fit her shoe budget. However, she has to choose between the boots and two pairs of high heels that together cost five dollars less than the boots. If one pair of heels costs $33 and the other costs twice as much, how many dollars are the boots?
104
nan
0
['```\ndef solution():\n    """Gloria is shoe shopping when she comes across a pair of boots that fit her shoe budget. However, she has to choose between the boots and two pairs of high heels that together cost five dollars less than the boots. If one pair of heels costs $33 and the other costs twice as much, how many dollars are the boots?"""\n    # Let\'s assign variables to the given values\n    boots_cost = x\n    heel1_cost = 33\n    heel2_cost = 2 * heel1_cost\n    total_heels_cost = heel1_cost + heel2_cost\n    total_cost_diff = total_heels_cost - boots_cost\n    # We know that total_cost_diff is 5 dollars less than boots_cost\n    # So, we can write an equation:

In [27]:
all_questions = df["input"].tolist()

In [28]:
all_sub_questions = df["subquestions"].tolist()

In [29]:
len(questions)

294

In [30]:
len(sub_questions)

294

In [31]:
final_questions = []

In [33]:
for question, sub_questions in zip(all_questions, all_sub_questions):
    reformed_question = "Question: " + question + "\n"
    for i, sub_question in enumerate(sub_questions):
        reformed_question = reformed_question + f"Sub-Question {i+1}. " + sub_question + "\n"
    final_questions.append(reformed_question)

In [41]:
print(final_questions[22])

Question: Ava and Emma want to know who is better at the new video game Ava got for her birthday. They are each going to play one level and whoever has the highest score wins. They receive 10 points for every enemy they jump on, 5 points for each berry they collect, and 30 points for every second left on the timer when they finish the level. If Ava jumps on 8 more enemies than Emma and collects 3 more berries, but finishes the level 4 seconds slower, what is the difference between their two scores?
Sub-Question 1. How many enemies did Emma jump on?
Sub-Question 2. How many berries did Ava collect?
Sub-Question 3. How many seconds did Ava take to finish the level?
Sub-Question 4. How many points did Ava score in total?
Sub-Question 5. How many points did Emma score in total?”
Sub-Question 6. What is the difference between their two scores?



In [55]:
new_df = pd.DataFrame(final_questions)

In [56]:
new_df = new_df.rename(columns={0:"input"})

In [58]:
df["new_input"] = final_questions

In [59]:
df

input  target  prev_answer  \
0    Gloria is shoe shopping when she comes across ...     104          NaN   
1    Marilyn's first record sold 10 times as many c...    8000          NaN   
2    Emily has 4 kids named Amy, Jackson, Corey, an...       4          NaN   
3    Jerome had 4 friends who came to visit him on ...     175          NaN   
4    Frankie watches TV after he finishes his homew...       3          NaN   
..                                                 ...     ...          ...   
289  A cup of mushrooms weighs 100 grams and has 3 ...      42         42.0   
290  While working at the restaurant, each of the f...    1520       1520.0   
291  John rents his car out 10 times a month for 3 ...     250        250.0   
292  Frankie and Binkie went bowling together.  Fra...     195        195.0   
293  My kitchen floor has a total area of 200 SqFt....    2400       2400.0   

     prev_score                                    prev_generation  \
0             0  [```\ndef solution():\n    """Gloria is shoe s...   
1             0  [We can solve this problem using algebra. Let'...   
2             0  [We can use algebra to solve this problem. Let...   
3             0  [We can solve this problem using Python as fol...   
4             0  [We can solve this problem by using algebra. L...   
..          ...                                                ...   
289           1  [```\ndef solution():\n    """A cup of mushroo...   
290           1  [```\ndef solution():\n    """While working at...   
291           1  [```\ndef solution():\n    """John rents his c...   
292           1  [```\ndef solution():\n    """Frankie and Bink...   
293           1  [```\ndef solution():\n    """My kitchen floor...   

                                       prev_code_error  \
0        cannot assign to operator (<string>, line 13)   
1                    invalid syntax (<string>, line 2)   
2                    invalid syntax (<string>, line 1)   
3    local variable 'friend4' referenced before ass...   
4                              No module named 'sympy'   
..                                                 ...   
289                                               None   
290                                               None   
291                                               None   
292                                               None   
293                                               None   

                                          subquestions  \
0    [How much do the two pairs of high heels cost ...   
1    [How many copies did Marilyn sell?, What is th...   
2    [How old is Corey?, How old is Amy?, How old i...   
3    [How many times did the second friend press th...   
4    [How many hours of TV did Frankie watch on Mon...   
..                                                 ...   
289  [How many cups of mushrooms does John eat per ...   
290  [How much money did Rafaela receive in tips?, ...   
291  [How much money does John make from renting ou...   
292  [How much higher than 90 was Frankie's score?,...   
293  [How many square tiles do I need to cover the ...   

                                             new_input  
0    Question: Gloria is shoe shopping when she com...  
1    Question: Marilyn's first record sold 10 times...  
2    Question: Emily has 4 kids named Amy, Jackson,...  
3    Question: Jerome had 4 friends who came to vis...  
4    Question: Frankie watches TV after he finishes...  
..                                                 ...  
289  Question: A cup of mushrooms weighs 100 grams ...  
290  Question: While working at the restaurant, eac...  
291  Question: John rents his car out 10 times a mo...  
292  Question: Frankie and Binkie went bowling toge...  
293  Question: My kitchen floor has a total area of...  

[294 rows x 8 columns]

In [60]:
df = df.drop(columns=["input"])

In [62]:
df = df.rename(columns={"new_input":"input"})

In [63]:
df

target  prev_answer  prev_score  \
0       104          NaN           0   
1      8000          NaN           0   
2         4          NaN           0   
3       175          NaN           0   
4         3          NaN           0   
..      ...          ...         ...   
289      42         42.0           1   
290    1520       1520.0           1   
291     250        250.0           1   
292     195        195.0           1   
293    2400       2400.0           1   

                                       prev_generation  \
0    [```\ndef solution():\n    """Gloria is shoe s...   
1    [We can solve this problem using algebra. Let'...   
2    [We can use algebra to solve this problem. Let...   
3    [We can solve this problem using Python as fol...   
4    [We can solve this problem by using algebra. L...   
..                                                 ...   
289  [```\ndef solution():\n    """A cup of mushroo...   
290  [```\ndef solution():\n    """While working at...   
291  [```\ndef solution():\n    """John rents his c...   
292  [```\ndef solution():\n    """Frankie and Bink...   
293  [```\ndef solution():\n    """My kitchen floor...   

                                       prev_code_error  \
0        cannot assign to operator (<string>, line 13)   
1                    invalid syntax (<string>, line 2)   
2                    invalid syntax (<string>, line 1)   
3    local variable 'friend4' referenced before ass...   
4                              No module named 'sympy'   
..                                                 ...   
289                                               None   
290                                               None   
291                                               None   
292                                               None   
293                                               None   

                                          subquestions  \
0    [How much do the two pairs of high heels cost ...   
1    [How many copies did Marilyn sell?, What is th...   
2    [How old is Corey?, How old is Amy?, How old i...   
3    [How many times did the second friend press th...   
4    [How many hours of TV did Frankie watch on Mon...   
..                                                 ...   
289  [How many cups of mushrooms does John eat per ...   
290  [How much money did Rafaela receive in tips?, ...   
291  [How much money does John make from renting ou...   
292  [How much higher than 90 was Frankie's score?,...   
293  [How many square tiles do I need to cover the ...   

                                                 input  
0    Question: Gloria is shoe shopping when she com...  
1    Question: Marilyn's first record sold 10 times...  
2    Question: Emily has 4 kids named Amy, Jackson,...  
3    Question: Jerome had 4 friends who came to vis...  
4    Question: Frankie watches TV after he finishes...  
..                                                 ...  
289  Question: A cup of mushrooms weighs 100 grams ...  
290  Question: While working at the restaurant, eac...  
291  Question: John rents his car out 10 times a mo...  
292  Question: Frankie and Binkie went bowling toge...  
293  Question: My kitchen floor has a total area of...  

[294 rows x 7 columns]

In [64]:
df.columns

Index(['target', 'prev_answer', 'prev_score', 'prev_generation',
       'prev_code_error', 'subquestions', 'input'],
      dtype='object')

In [66]:
cols = ['input','target', 'prev_answer', 'prev_score', 'prev_generation',
       'prev_code_error', 'subquestions']

In [67]:
df = df[cols]

In [68]:
df

input  target  prev_answer  \
0    Question: Gloria is shoe shopping when she com...     104          NaN   
1    Question: Marilyn's first record sold 10 times...    8000          NaN   
2    Question: Emily has 4 kids named Amy, Jackson,...       4          NaN   
3    Question: Jerome had 4 friends who came to vis...     175          NaN   
4    Question: Frankie watches TV after he finishes...       3          NaN   
..                                                 ...     ...          ...   
289  Question: A cup of mushrooms weighs 100 grams ...      42         42.0   
290  Question: While working at the restaurant, eac...    1520       1520.0   
291  Question: John rents his car out 10 times a mo...     250        250.0   
292  Question: Frankie and Binkie went bowling toge...     195        195.0   
293  Question: My kitchen floor has a total area of...    2400       2400.0   

     prev_score                                    prev_generation  \
0             0  [```\ndef solution():\n    """Gloria is shoe s...   
1             0  [We can solve this problem using algebra. Let'...   
2             0  [We can use algebra to solve this problem. Let...   
3             0  [We can solve this problem using Python as fol...   
4             0  [We can solve this problem by using algebra. L...   
..          ...                                                ...   
289           1  [```\ndef solution():\n    """A cup of mushroo...   
290           1  [```\ndef solution():\n    """While working at...   
291           1  [```\ndef solution():\n    """John rents his c...   
292           1  [```\ndef solution():\n    """Frankie and Bink...   
293           1  [```\ndef solution():\n    """My kitchen floor...   

                                       prev_code_error  \
0        cannot assign to operator (<string>, line 13)   
1                    invalid syntax (<string>, line 2)   
2                    invalid syntax (<string>, line 1)   
3    local variable 'friend4' referenced before ass...   
4                              No module named 'sympy'   
..                                                 ...   
289                                               None   
290                                               None   
291                                               None   
292                                               None   
293                                               None   

                                          subquestions  
0    [How much do the two pairs of high heels cost ...  
1    [How many copies did Marilyn sell?, What is th...  
2    [How old is Corey?, How old is Amy?, How old i...  
3    [How many times did the second friend press th...  
4    [How many hours of TV did Frankie watch on Mon...  
..                                                 ...  
289  [How many cups of mushrooms does John eat per ...  
290  [How much money did Rafaela receive in tips?, ...  
291  [How much money does John make from renting ou...  
292  [How much higher than 90 was Frankie's score?,...  
293  [How many square tiles do I need to cover the ...  

[294 rows x 7 columns]

In [70]:
df.to_json(r"C:\Users\kaush\Desktop\fuck_you_chatPAL\pal\datasets\gsmr_with_subquestions.jsonl", orient="records", lines=True)

## Results:

In [8]:
RESULTS = r"C:\Users\kaush\Desktop\fuck_you_chatPAL\pal\eval_results\24-04-2023_231642_kaush_gsmr_with_subquestions.chat.jsonl"

In [9]:
results = pd.read_json(RESULTS, lines=True)

ValueError: Expected object or value

In [81]:
results["score"].value_counts()[1]/len(results["score"])

0.5347985347985348

    Not all questions were completed, only some 273 were out of 294.
    On the ones that were completed, the accuracy was 53.47%.

    Let's get all the last # Sub-Question Answer: Numbers from each question and see if we can get a better accuracy.

In [2]:
print(results["generation"][0][0])

NameError: name 'results' is not defined

In [105]:
import re
txt = results["generation"][0][0]
x = re.findall("# Sub-Question Answer: \d+", txt)
x = x[-1]
x = re.findall("\d+", x)[0]

In [114]:
def obtain_sub_question_answer(text):
    text = text[0]
    x = re.findall("# Sub-Question Answer: \d+", text)
    if len(x) > 0:
        x = x[-1]
        x = re.findall("\d+", x)[0]
    else:
        x = 0   
    return int(x)

In [115]:
results["implicit_self_eval_ans"] = results["generation"].map(obtain_sub_question_answer)

In [123]:
self_eval_boolean_result = (results["implicit_self_eval_ans"] == results["target"])

In [128]:
len(results["score"])

273

In [133]:
self_eval_boolean_result.value_counts()[True]/len(results["score"])

0.4908424908424908

In [129]:
results["score"].value_counts()[1]/len(results["score"])

0.5347985347985348

In [140]:
results["implicit_self_eval_ans"].astype(float)

0        94.0
1      8000.0
2        11.0
3       175.0
4         3.0
        ...  
268      80.0
269     265.0
270     540.0
271       4.0
272      70.0
Name: implicit_self_eval_ans, Length: 273, dtype: float64

In [142]:
self_eval_comparison = results["answer"] == results["implicit_self_eval_ans"].astype(float)

In [143]:
self_eval_comparison.value_counts()

True     179
False     94
dtype: int64

In [144]:
179/(179+94)

0.6556776556776557

In [149]:
print(results["generation"][0][0])

```
def solution():
    # How much do the two pairs of high heels cost in total?
    heel1_cost = 33
    heel2_cost = heel1_cost * 2
    total_heels_cost = heel1_cost + heel2_cost
    # Sub-Question Answer: 99

    # How much does the more expensive pair of high heels cost?
    expensive_heel_cost = heel2_cost
    # Sub-Question Answer: 66

    # What is the cost of the boots?
    boots_cost = total_heels_cost - 5
    # Sub-Question Answer: 94

    # How many dollars are the boots?
    result = boots_cost
    # Sub-Question Answer: 94
    
    return result
```


In [148]:
print(results["input"][0])

Question: Gloria is shoe shopping when she comes across a pair of boots that fit her shoe budget. However, she has to choose between the boots and two pairs of high heels that together cost five dollars less than the boots. If one pair of heels costs $33 and the other costs twice as much, how many dollars are the boots?
Sub-Question 1. How much do the two pairs of high heels cost in total?
Sub-Question 2. How much does the more expensive pair of high heels cost?
Sub-Question 3. What is the cost of the boots?
Sub-Question 4. How many dollars are the boots?



In [150]:
Let's answer each sub-question in a separate code block initially and insert a comment containing the sub-question answer as- Sub-Question Answer:answer value.
Question: Gloria is shoe shopping when she comes across a pair of boots that fit her shoe budget. However, she has to choose between the boots and two pairs of high heels that together cost five dollars less than the boots. If one pair of heels costs $33 and the other costs twice as much, how many dollars are the boots?
Sub-Question 1. How much do the two pairs of high heels cost in total?
# How much do the two pairs of high heels cost in total?
heel1_cost = 33
heel2_cost = heel1_cost * 2
total_heels_cost = heel1_cost + heel2_cost
# Sub-Question Answer: 99
Check the work on the sub-question above. Ensure there are no logic errors or coding errors, keeping in mind the following:
Question: Gloria is shoe shopping when she comes across a pair of boots that fit her shoe budget. However, she has to choose between the boots and two pairs of high heels that together cost five dollars less than the boots. If one pair of heels costs $33 and the other costs twice as much, how many dollars are the boots?
Answer "True" if sure, else "False".

SyntaxError: EOL while scanning string literal (4090270993.py, line 1)

In [151]:
Sub-Question 2. How much does the more expensive pair of high heels cost?
expensive_heel_cost = heel2_cost
# Sub-Question Answer: 66
Check the work on the sub-question above. Ensure there are no logic errors or coding errors, keeping in mind the following:
Question: Gloria is shoe shopping when she comes across a pair of boots that fit her shoe budget. However, she has to choose between the boots and two pairs of high heels that together cost five dollars less than the boots. If one pair of heels costs $33 and the other costs twice as much, how many dollars are the boots?
Answer "True" if sure, else "False".

SyntaxError: invalid syntax (3772745440.py, line 4)

: 

In [ ]:
Sub-Question 3. What is the cost of the boots?
boots_cost = total_heels_cost - 5
# Sub-Question Answer: 94
Check the work on the sub-question above. Ensure there are no logic errors or coding errors, keeping in mind the following:
Question: Gloria is shoe shopping when she comes across a pair of boots that fit her shoe budget. However, she has to choose between the boots and two pairs of high heels that together cost five dollars less than the boots. If one pair of heels costs $33 and the other costs twice as much, how many dollars are the boots?
Answer "True" if sure, else "False".

### Randomly picking 5 results which are wrong:

In [1]:
results

NameError: name 'results' is not defined